allennlp/models/language_model.py

```diff
@@ -100,22 +120,88 @@ def __init__(self,
         super().__init__(vocab)
         self._text_field_embedder = text_field_embedder
 
-        if contextualizer.is_bidirectional() is not bidirectional:
+        # Only true when contextualizer is non-None and bidirectional is True
+        self._use_contextualizer_arg = False
+        if contextualizer is not None and (forward_contextualizer is not None or
+                                           backward_contextualizer is not None):
             raise ConfigurationError(
-                    "Bidirectionality of contextualizer must match bidirectionality of "
-                    "language model. "
-                    f"Contextualizer bidirectional: {contextualizer.is_bidirectional()}, "
-                    f"language model bidirectional: {bidirectional}")
-
-        self._contextualizer = contextualizer
+                    "Cannot provide both contextualizer and either "
+                    "forward_contextualizer or backward_contextualizer.")
+
+        if contextualizer is not None:
+            warnings.warn("``contextualizer`` was deprecated in version 0.8.2 . It was "
+                          "replaced with two more flexible arguments: "
+                          "``forward_contextualizer`` and ``backward_contextualizer``. "
+                          "It will be removed in version 0.10 .",
+                          DeprecationWarning)
+            if contextualizer.is_bidirectional() is not bidirectional:
+                raise ConfigurationError(
+                        "Bidirectionality of contextualizer must match bidirectionality "
+                        "of language model. "
+                        f"Contextualizer bidirectional: {contextualizer.is_bidirectional()}, "
+                        f"language model bidirectional: {bidirectional}")
+            if contextualizer.is_bidirectional():
+                # TODO (nfliu): Emit warning about cheating
```

Comment:
not too sure what the wording of this warning would be like, thoughts @matt-peters ?